### ISSUE TITLE
[CAL-3995] Error Could not book the meeting. Unexpected token '<', "<!DOCTYPE "... is not valid JSON

### ISSUE THREAD

**Issue by joseuo94** [calcom/cal.com#15655]
A lot of customers reaching out with this issue.

It still creates the booking on [Cal.com](http://Cal.com) but does not send the confirmation email

if we open the booking at [https://app.cal.com/bookings/upcoming](https://app.cal.com/bookings/upcoming)
it says the Conferencing link is in the confirmation email

![image.png](https://uploads.linear.app/e86bf957-d82f-465e-b205-135559f4b623/7686ed83-d05e-4ec5-b700-b0d43712a67f/016ac9aa-2b62-435e-9891-7bd93e106e2c?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2U4NmJmOTU3LWQ4MmYtNDY1ZS1iMjA1LTEzNTU1OWY0YjYyMy83Njg2ZWQ4My1kMDVlLTRlYzUtYjcwMC1iMGQ0MzcxMmE2N2YvMDE2YWM5YWEtMmI2Mi00MzVlLTk4OTEtN2JkOTNlMTA2ZTJjIiwiaWF0IjoxNzIwMDE1ODYwLCJleHAiOjE3NTE1NTE4NjB9.a8zPtKFz6NJ15wWcxiqBXxNHwhYlUF7eJozuNMzsgkE)

Could not book the meeting.
Unexpected token '<', "<!DOCTYPE "... is not valid JSON

![image.png](https://uploads.linear.app/e86bf957-d82f-465e-b205-135559f4b623/34e935e2-075c-4352-99e6-9e4bc56e49bf/82bcba2c-3a16-43aa-8425-e00bf0fc4d48?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2U4NmJmOTU3LWQ4MmYtNDY1ZS1iMjA1LTEzNTU1OWY0YjYyMy8zNGU5MzVlMi0wNzVjLTQzNTItOTllNi05ZTRiYzU2ZTQ5YmYvODJiY2JhMmMtM2ExNi00M2FhLTg0MjUtZTAwYmYwZmM0ZDQ4IiwiaWF0IjoxNzIwMDE1NDcyLCJleHAiOjE3NTE1NTE0NzJ9.oLrPItqh66Ubgebu7Z_B3mM2nR7CoLZelCnnKTiSwKU)

![image.png](https://uploads.linear.app/e86bf957-d82f-465e-b205-135559f4b623/5757a38d-ccf0-4bcd-8e49-1733d9b356c6/7439f87b-fa64-459e-b213-cc680caa928e?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2U4NmJmOTU3LWQ4MmYtNDY1ZS1iMjA1LTEzNTU1OWY0YjYyMy81NzU3YTM4ZC1jY2YwLTRiY2QtOGU0OS0xNzMzZDliMzU2YzYvNzQzOWY4N2ItZmE2NC00NTllLWIyMTMtY2M2ODBjYWE5MjhlIiwiaWF0IjoxNzIwMDE1NDcyLCJleHAiOjE3NTE1NTE0NzJ9.w4P5_ymA8R7NEOvigXAdFVgShRaZcj36Kq6jSPCC9q0)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3995](https://linear.app/calcom/issue/CAL-3995/error-could-not-book-the-meeting-unexpected-token-doctype-is-not-valid)</sub>

**Comment by sujal12344** [comment]
It take effort, can you give bounty on this issue.

**Comment by fabio0296** [comment]
Isn't this realted to https://github.com/calcom/cal.com/issues/15644?

**Comment by yalla-habibi** [comment]
@prynsh In my case this happens, when there is a booking page for a group event (in my case it's a round robin). When all required fields a re filled, and the button Confirm is clicked, it takes around 20-30 seconds of loading, and finally the error presented above is displayed, and no redirection is performed. However, the event seems to be created - I can see on the admin panel, and in the calendar (both organiser and atendee). But no workflow is being triggered, effectively no email and SMS in my case is not being sent. 

@fabio0296 yes that the same issue 

Please let me know guys when the fix is ready, our sale team is suffering a lot, they are literally blocked for two last days. All integrations (CRM, Communication, ERP) is dead when the cal.com doesnt work as supposed to.

**Comment by prynsh** [comment]
@yalla-habibi I am still unable to understand. I am attaching the pictures below for your reference.
This is the image of the dashboard 

![image](https://github.com/calcom/cal.com/assets/81437154/3ac9e515-f5a1-46f3-9a67-aa4f0a3520ba)
Next when going to the bookings page, I do not see any bookings.

![image](https://github.com/calcom/cal.com/assets/81437154/78bc68c3-989d-4bec-a952-33282fad4f13)

**Comment by emrysal** [comment]
@yalla-habibi could you send me the event type where this happens? Having a bit of an issue replicating so it's probably due to a setting of sorts.

**Comment by yalla-habibi** [comment]
Uploading Screenshare - 2024-07-08 11_46_38 AM.mp4…

**Comment by yalla-habibi** [comment]
@emrysal @raza001 @prynsh Please find above a screenshare. @emrysal it's as I outlined above, it's a group event with round robin algorithm

**Comment by yalla-habibi** [comment]
https://github.com/calcom/cal.com/assets/25034260/a7fc2a47-ee99-4cc5-800a-e911e161e163

**Comment by milospuac** [comment]
Can you all check if you have any CRM apps enabled for that event? If yes, try disabling it and check again. I think this is a due to a faulty app that's enabled for that event

**Comment by bdzb** [comment]
now the bug is gone @yalla-habibi 

Uploading Recording 2024-07-08 200112.mp4…

**Comment by prynsh** [comment]
Yes, I believe the same because when I was doing it locally I faced no errors and was redirected to confirmation  page.

**Comment by raza001** [comment]
Same here

**Comment by fabio0296** [comment]
Same here, couldn't reproduce the issue on my local

**Comment by yalla-habibi** [comment]
@yalla-habibi guys can you check it on web application and not local? use link: 

https://cal.com/team/pricer/badanie-rynku-lokalnego

@milospuac thank for this lead, but I have no applications enabled:
<img width="1437" alt="image" src="https://github.com/calcom/cal.com/assets/25034260/97b4f516-ef86-4699-bfb5-6ff8aa9bbd11">

(however I did try 2 months ago to enable pipedrive CRM, but at this moment it is not installed )

**Comment by yalla-habibi** [comment]
Guys could you take a look if it's anyhow connected with the issue? It keeps on appearing when I close the settings of the event:
<img width="1439" alt="image" src="https://github.com/calcom/cal.com/assets/25034260/119eb572-ed03-42bf-ab6b-f07b0f7058bb">
It didnt use to be a case before the error

**Comment by yalla-habibi** [comment]
@fabio0296 @raza001 @prynsh @bdzb @emrysal @milospuac any updates on this? We are suffering with the bug for more than 7 working days, this is killing our automated sales process

**Comment by emrysal** [comment]
@yalla-habibi We've diagnosed this issue down to slow execution time, some kind of configuration or external connection causes a long time for the function to execute - when this exceeds 1 minute CloudFlare cuts it off, and returns a "timeout" error. We haven't yet figured out why you guys have this issue (CRM connection, something else?) - and this is hard to investigate as the code is cut-off mid execution, we can see no errors as a result.

**Comment by yalla-habibi** [comment]
@emrysal so what is the status? As for the moment, the error has disappeared however I understand that it's not due to your work, what makes me feel like it's not a very stable situation for us, am I correct?

**Comment by piannaf** [comment]
Seeing this issue again 
![image](https://github.com/user-attachments/assets/3d365569-abda-4c34-832f-01a76cf34cbd)

**Comment by subwaysurety** [comment]
I can confirm the same issue with Cloudflare. @emrysal any ETA when this will be addressed?

**Comment by Cluster2a** [comment]
We are running cal.com in a docker container. There is no documentation for this environment variable:
https://github.com/calcom/docker

**Comment by subwaysurety** [comment]
> We are running cal.com in a docker container. There is no documentation for this environment variable: https://github.com/calcom/docker

see my comment above it's a massive core issue they need to fix, docker or not it doesn't matter

**Comment by kukovinetsyevhenii** [comment]
Hi
I am trying to use cal.com api
curl --location 'https://api.cal.com/v1/bookings?apiKey=cal_live_5a00f556540e06cfc5b852016ed7d641' \
--header 'Content-Type: application/json' \
--data-raw '{
  "eventTypeId": 952631,
  "start": "2024-08-26T09:00:00.000Z",
  "responses": {
    "name": "John Doe",
    "email": "yevhehii250@gmail.com",
    "guests": [],
    "location": {
      "value": "inPerson",
      "optionValue": ""
    }
  },
  "metadata": {},
  "timeZone": "America/New_York",
  "language": "en",
  "title": "Appointment with kendall office with John Doe",
  "description": null,
  "status": "PENDING",
  "smsReminderNumber": null
}
'


I got error

{
    "message": "no_available_users_found_error"
}


That happens 100% even though I control all parameters.
Never success.
Please let me know what is the issue.

**Comment by subwaysurety** [comment]
@emrysal is this finally getting fixed in 4.5 sprint ? It makes app unusable

**Comment by girassolar** [comment]
same problem here. The metting is scheduled, I even get the email, but on n8n, it says: "no_available_users_found_error".

**Comment by Hamzaclaus** [comment]
Has anyone fixed the problem? We are having the same issue now.

**Comment by subwaysurety** [comment]
I don't think they give a flying F tbh at this point. This makes tool unusable and they keep pushing it further in backlog.

**Comment by Hamzasoinq** [comment]
Guys that’s pathetic it’s almost 2 months still people not able to use to freaking book calls properly I understand if a bug that does not affect but it freaking prevent from booking needs to be sorted yesterday

**Comment by subwaysurety** [comment]
@keithwillcode, what are the chances this issue is getting fixed by the end of the year?

**Comment by keithwillcode** [comment]
@subwaysurety in terms of fixing any booking errors like this that result in the “blue booking error”, we have an internal goal of completely ridding them from the system by EOY. 

@emrysal is heading this up.

**Comment by subwaysurety** [comment]
> @subwaysurety it’s absolutely not true that we don’t care. Sorry you feel that way. 🙏🏼
> 
> @Hamzaclaus we don’t have an issue where hosters of the app can’t accept bookings. We have many folks using the app fine.
> 
> If it’s an env problem, what do you want the fix to be?

I appreciate your reply, but this is unusable as a hosted version. Although the issue has been identified a while ago.

It's literally same error, I would gladly fix it with a duct tape and wd40 but it's something in the core, so none of us can't and yes it's freaking frustrating when a core functionality doesn't work and turn this beautiful product in a hot pile of garbage.

**Comment by keithwillcode** [comment]
@subwaysurety I understand your frustration but as I mentioned, it’s not unusable. We have many folks using it just fine and with great success. 

If you take a moment to share a few more details and with positive vibes only, we’ll be happy to help try and figure out what’s going on. Feel free to book a call with me at https://i.cal.com/keith and we can walk through the issue together.

**Comment by Cluster2a** [comment]
@keithwillcode, this is a confirmed issue and it even apears on your provided docker environment (https://github.com/calcom/docker).

You are addressing this issue from one release to anoter (4.3 / 4.4 / 4.5) and after 4 months you are asking for details?

We also had to move away from calcom, as this solution seems not to be stable at all (at least for the hosted version).

**Comment by subwaysurety** [comment]
> @subwaysurety I understand your frustration but as I mentioned, it’s not unusable. We have many folks using it just fine and with great success.
> 
> If you take a moment to share a few more details and with positive vibes only, we’ll be happy to help try and figure out what’s going on. Feel free to book a call with me at https://i.cal.com/keith and we can walk through the issue together.

I appreciate your willingness to have a call, but let's focus on the issue first. 
As @Cluster2a mentioned, it doesn't work with the docker env you provided. 

It would make more sense if your team could provide a workaround, if there is one, to get the bloody thing from the grave.

**Comment by keithwillcode** [comment]
> @keithwillcode, this is a confirmed issue and it even apears on your provided docker environment (https://github.com/calcom/docker).
> 
> You are addressing this issue from one release to anoter (4.3 / 4.4 / 4.5 / 4.7) and after 4 months you are asking for details?
> 
> We also had to move away from calcom, as this solution seems not to be stable at all (at least for the hosted version).

As noted in the Docker repo, we don’t officially support the Docker installation. Hence, the solution times can often be slower. 

We plan to officially support it by EOY and it likely will be rolled into the main repo.

**Comment by subwaysurety** [comment]
> > @keithwillcode, this is a confirmed issue and it even apears on your provided docker environment (https://github.com/calcom/docker).
> > You are addressing this issue from one release to anoter (4.3 / 4.4 / 4.5 / 4.7) and after 4 months you are asking for details?
> > We also had to move away from calcom, as this solution seems not to be stable at all (at least for the hosted version).
> 
> As noted in the Docker repo, we don’t officially support the Docker installation. Hence, the solution times can often be slower.
> 
> We plan to officially support it by EOY and it likely will be rolled into the main repo.

so regular no docker is working fine?

**Comment by keithwillcode** [comment]
> > > @keithwillcode, this is a confirmed issue and it even apears on your provided docker environment (https://github.com/calcom/docker).
> > > You are addressing this issue from one release to anoter (4.3 / 4.4 / 4.5 / 4.7) and after 4 months you are asking for details?
> > > We also had to move away from calcom, as this solution seems not to be stable at all (at least for the hosted version).
> > 
> > 
> > As noted in the Docker repo, we don’t officially support the Docker installation. Hence, the solution times can often be slower.
> > We plan to officially support it by EOY and it likely will be rolled into the main repo.
> 
> so regular no docker is working fine?

Yes it’s how our SaaS app runs. 

In some edge cases, this error pops up but we try and address them as they come up. That’s the project I mentioned @emrysal is heading up.

**Comment by subwaysurety** [comment]
@keithwillcode you mean walking away from Docker version?

**Comment by keithwillcode** [comment]
No. Just saying that in our SaaS app deployment, we don't use Docker. We don't plan to move away from Docker though. We plan to support it better moving forward in 2025.

**Comment by animoller** [comment]
Just got this error with a client. All the sessions were booked but they got this token error and couldn't proceed to the confirmation page.

**Comment by Cluster2a** [comment]
WOW - fixing bugs is not planned...

**Comment by PeerRich** [comment]
this was closed as it was fixed

do you still experience the issue? can you share a video?

**Comment by Cluster2a** [comment]
@PeerRich, me migrated to another tool, because of this issue. The issue is cloed with marked as "Not planned" and 15930 which seemd like a duplicate is still open. So for me it was not clear that it is fixed by now.
